torchdata/datapipes/iter/util/combining.py

```diff
@@ -213,3 +216,139 @@ def __iter__(self) -> Iterator:
 
     def __len__(self) -> int:
         return len(self.source_iterdatapipe)
+
+
+def _drop_index(idx_data):
+    idx, data = idx_data
+    return data
+
+
+@functional_datapipe("round_robin_demux")
+class RoundRobinDemultiplexerIterDataPipe(IterDataPipe):
+    r"""
+    Splits the input DataPipe into multiple child DataPipes in the round-robin order (functional name: ``round_robin_demux``).
+    A list of the child DataPipes is returned from this operation.
+
+    Args:
+        datapipe: Iterable DataPipe being filtered
+        num_instances: number of instances of the DataPipe to create
+        buffer_size: this defines the maximum number of inputs that the buffer can hold across all child
+            DataPipes while waiting for their values to be yielded.
+            Defaults to ``1000``. Use ``-1`` for the unlimited buffer.
+
+    Examples:
+        >>> from torchdata.datapipes.iter import IterableWrapper
+        >>> def odd_or_even(n):
```

Comment:
this doesn't seem to be used in this example?

Reply:
Good catch.